test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      set = testSet.setByType(type).set;
+
+      expect(set.contains(values.first), false);
+
+      realm.write(() {
+        set.add(values.first);
+      });
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      set = testSet.setByType(type).set;
+      expect(set.contains(values.first), true);
+    });
+
+    test('RealmSet<$type> add', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      final testSet = TestRealmSets(1);
+
+      var values = testSet.values(type);
+
+      realm.write(() {
+        realm.add(testSet);
+        var set = testSet.setByType(type).set;
+        set.add(values.first);
+      });
+
+      var set = testSet.setByType(type).set;
+      // values = testSet.values(type);
+      values = testSet.getValuesOrManagedValues(realm, type);
+
+      expect(set.contains(values.first), true);
+    });
+
+    test('RealmSet<$type> remove', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      realm.write(() {
+        set.add(values.first);
+      });
+
+      expect(set.length, 1);
+
+      realm.write(() {
+        expect(set.remove(values.first), true);
+      });
+
+      expect(set.length, 0);
+    });
+
+    test('RealmSet<$type> length', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      final testSet = TestRealmSets(1);
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      var set = testSet.setByType(type).set;
+      var values = testSet.values(type);
+
+      expect(set.length, 0);
+
+      realm.write(() {
+        set.add(values.first);
+      });
+
+      expect(set.length, 1);
+
+      realm.write(() {
+        for (var value in values) {
+          set.add(value);
+        }
+      });
+
+      expect(set.length, values.length);
+    });
+
+    test('RealmSet<$type> elementAt', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      final testSet = TestRealmSets(1);
+      var values = testSet.values(type);
+
+      realm.write(() {
+        realm.add(testSet);
+        var set = testSet.setByType(type).set;
+        set.add(values.first);
+      });
+
+      var set = testSet.setByType(type).set;
+
+      expect(() => set.elementAt(-1), throws<RealmException>("Index out of range"));
+      expect(() => set.elementAt(800), throws<RealmException>());
```

Comment:
Why don't we check the exception string here?

Reply:
fixed